### DISCUSSION TITLE
Welcome to serverless-appsync-simulator Discussions!

### DISCUSSION THREAD

**Discussion by bboure** [bboure/serverless-appsync-simulator#61] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by j0t1** [comment]
Hi There, 

Can you please confirm if Javascipt resolvers supported by this plugin? If yes, can you please share an example?

Thanks !

**Comment by bboure** [comment]
Hi,

This plugin does not support JS resolvers (yet). 
For various reasons:
- I haven't had time to maintain it lately, and I have stopped using it
- JS resolvers are only available on the [v2 of the serverless-appsync-plugin](https://github.com/sid88in/serverless-appsync-plugin/tree/alpha) for now

I doubt I will ever port v2 to this plugin. Instead, my goal is to improve the DX in the `serverless-appsync-plugin` plugin itself, and remove the necessity for local emulation. i.e. I want to shorten the deployment time so you can test your code directly in the cloud.

**Reply by j0t1** [reply]
Thanks for your response @bboure.  Do you suggest any other plugin-in/library to test appsync JS resolvers on local machine (without having to deploy in cloud)?

**Reply by bboure** [reply]
I can offer you two options:

1. There is a CLI command to simulate/test Resolvers.

Read [this article](https://aws.amazon.com/blogs/aws/aws-appsync-graphql-apis-supports-javascript-resolvers/), check the "Testing a resolver" section. 

2. I am currently building a product that does the above, but provides a simple GUI to help you: [GraphBolt](https://graphbolt.dev/)

It is still in the private beta phase, but it will (very) soon be in public beta.

**Reply by bboure** [reply]
important side note: none of the above solutions execute resolvers against a data source. They only provide a way to check the resulting JSON that will be sent to the datasource.
